types/sharedb/lib/sharedb.d.ts

```diff
@@ -209,6 +216,31 @@ export interface ClientRequest {
     [propertyName: string]: any;
 }
 
+export interface AnyDataObject {
+    [key: string]: any;
+}
+
+export interface ServerResponseSuccess {
+    a?: RequestAction;
+    c?: CollectionName;
+    d?: DocumentID;
+    extra?: number;
```

Comment:
The `extra` here could be anything, it's so that a custom query implementation or middleware can send whatever it wants down, since the normal query results must be an array of documents.